### PR TITLE
feat(icon-button): add styled-system

### PR DESCRIPTION
**Pull request by dawidzarzycki**
### Proposed behaviour
add styled-system

### Current behaviour
there is no support for styled-system props

### Checklist

- [x] Commits follow our style guide
- [ ] <del> Screenshots are included in the PR if useful
- [ ] <del> All themes are supported if required
- [ ]  <del> Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] <del> Typescript `d.ts` file added or updated if required
- [ ] <del> Carbon implementation and Design System documentation are congruent

### Additional context
d.ts file be added in a different PR

### Testing instructions
https://codesandbox.io/s/carbon-quickstart-forked-zszrd?file=/src/index.js

### CODE REVIEW

File: src/components/icon-button/icon-button.component.js [hunk 1]
```diff
@@ -4,31 +4,34 @@ import Events from "../../utils/helpers/events";
...
+        p={p}
+      >
+        {children}
+      </StyledIconButton>
+    );
+  }
+);
 
 IconButton.propTypes = {
   /** Children prop is restricted to one Icon Component */
```
Comment (sjohnson-sage) [inline src/components/icon-button/icon-button.component.js:37]: You need to add the margin props into the propTypes (and not the padding ones)

File: src/components/icon-button/icon-button.style.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import styled, { css } from "styled-components";
+import { space } from "styled-system";
```
Comment (sjohnson-sage) [inline src/components/icon-button/icon-button.style.js:2]:
```suggestion
import { margin } from "styled-system";
```

We decided in the Jira ticket to only add margin options to this component

File: src/components/icon-button/icon-button.style.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 import StyledIcon from "../icon/icon.style";
 import { baseTheme } from "../../style/themes";
 
 const StyledIconButton = styled.button.attrs({ type: "button" })`
   ${({ theme, disabled }) => css`
+    ${space}
     background: transparent;
     border: none;
-    padding: 0;
```
Comment (sjohnson-sage) [inline src/components/icon-button/icon-button.style.js:9]: Keep the padding as 0, we don't want the option to override it here

**Review by sjohnson-sage — COMMENTED** [review]
We only want the margin props as it says on the Jira ticket, see https://github.com/Sage/carbon/blob/master/src/components/pill/pill.style.js as an example of how to do this

File: src/components/icon-button/icon-button.style.js [hunk 4]
```diff
@@ -1,12 +1,13 @@
 import styled, { css } from "styled-components";
+import { space } from "styled-system";
 import StyledIcon from "../icon/icon.style";
 import { baseTheme } from "../../style/themes";
 
 const StyledIconButton = styled.button.attrs({ type: "button" })`
   ${({ theme, disabled }) => css`
+    ${space}
```
Comment (sjohnson-sage) [inline src/components/icon-button/icon-button.style.js:8]:
```suggestion
    ${margin}
```

File: src/components/icon-button/icon-button.component.js [hunk 5]
```diff
@@ -37,6 +37,8 @@ IconButton.propTypes = {
   }).isRequired,
   /** Callback */
   onAction: PropTypes.func.isRequired,
+  /** add margin to component */
+  margin: PropTypes.string,
```
Comment (edleeks87) [inline src/components/icon-button/icon-button.component.js:41]: https://github.com/Sage/carbon/blob/c615538d5ed2b812dc81fb4227e07b54d36eeb57/src/style/utils/filter-styled-system-margin-props.js

I think we should use this to define the margin prop types

**Review by sjohnson-sage — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**